ruler-cli/src/main/java/com/spotify/ruler/cli/RulerCli.kt

```diff
@@ -0,0 +1,118 @@
+@file:OptIn(ExperimentalSerializationApi::class)
+
+package com.spotify.ruler.cli
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+import com.spotify.rulercommon.BaseRulerTask
+import com.spotify.rulercommon.FEATURE_NAME
+import com.spotify.rulercommon.dependency.ArtifactResult
+import com.spotify.rulercommon.dependency.DependencyComponent
+import com.spotify.rulercommon.dependency.DependencyEntry
+import com.spotify.rulercommon.dependency.DependencySanitizer
+import com.spotify.rulercommon.dependency.JarArtifactParser
+import com.spotify.rulercommon.models.AppInfo
+import com.spotify.rulercommon.models.DeviceSpec
+import com.spotify.rulercommon.models.RulerConfig
+import com.spotify.rulercommon.sanitizer.ClassNameSanitizer
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+import java.io.File
+
+class RulerCli : CliktCommand(), BaseRulerTask {
+    private val dependencyMap by option().file().required()
+    private val rulerConfigJson by option().file().required()
+    private val apkFile by option().file().required()
+    private val reportDir by option().file(canBeDir = true).required()
+    private val mappingFile: File? by option().file()
+
+    override fun print(content: String) = echo(content)
+
+    override fun provideMappingFile() = mappingFile
+
+    override fun provideResourceMappingFile(): File? {
+        return null
+        TODO("Not yet implemented")
```

Comment:
We should add the resource mapping file as an optional argument

Reply:
done